astropy/nddata/tests/test_nddata.py

```diff
@@ -422,23 +422,49 @@ def test_nddata_str():
          [[5 6]
           [7 8]]]"""[1:])
 
+    # let's add units!
+    arr = NDData(np.array([1, 2, 3]), unit="km")
+    assert str(arr) == '[1 2 3] km'
```

Comment:
Might also want to test for more complex using like "erg cm^-2 s^-1 A^-1" 😬

Reply:
Thanks! will do.

Reply:
Done!